File: lexilla/test/examples/ruby/AllStyles.rb
# Enumerate all styles where possible: 0..31,40..45
# 30,31,40,45 are never set and 1 switches rest of file to error state

#0 whitespace
    #
	#

#1:error, can be set with a heredoc delimiter >256 characters but that can't be recovered from
#<<ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789

#2:comment line

#3:POD
=begin
3:POD
=end

#4:number
4

#5:word
super

#6:string
"6:double quotes"

#7:single quoted string
'7:single quotes'

#8:class name
class ClassName end

#9:def name
def Function end

#10:operator
&

#11:identifier
identifier

#12:regex
/[12a-z]/

#13:global
$global13

#14:symbol
:symbol14

#15:module name
module Module15 end

#16:instance var
@instance16

#17:class var
@@class17

#18:back ticks
`18`

#19:data section at end of file

#20:here delimiter
<<DELIMITER20
DELIMITER20

#21:single quoted heredoc
<<'D'
21:here doc #{1 + 1}
D

#22:double quoted heredoc
<<"D"
22:here doc #{1 + 1}
D

#23:back tick quoted heredoc
<<`D`
23:here doc #{1 + 1}
D

#24:q quoted string
%q!24:quotes's!

#25:Q quoted string
%Q!25:quotes"s!

#26:executed string
%x(echo 26)

#27:regex
%r(27[a-z]/[A-Z]+)

#28:interpolable string array
%W(28 cgi.rb complex.rb date.rb #{1} )

#29:demoted keyword do
while 1 do end

# 30,31,40,45 never set

#41:non-interpolable string array
%w(#{1 + 1})

#42:non-interpolable symbol array
%i(#{1 + 1})

#43:interpolable symbol array
%I(#{1 + 1})

#44:symbol
%s(#{1 + 1})

#19:data section
__END__

File: lexilla/test/examples/lua/AllStyles.lua
-- Enumerate all styles: 0 to 20
-- 3 (comment doc) is not currently produced by lexer

--[[ comment=1 ]]

--[[ whitespace=0 ]]
	-- w

-- comment line=2

--- comment doc=3
-- still comment doc

-- still comment doc
3	-- comment doc broken only by code

-- number=4
37

-- keyword=5
local a

-- double-quoted-string=6
"str"

-- single-quoted-string=7
'str'

-- literal string=8
[[ literal ]]

-- unused preprocessor=9
$if

-- operator=10
*

-- identifier=11
identifier=1

-- string EOL=12
"unclosed

-- keyword 2=13
print

-- keyword 3=14
keyword3

-- keyword 4=15
keyword4

-- keyword 5=16
keyword5

-- keyword 6=17
keyword6

-- keyword 7=18
keyword7

-- keyword 8=19
keyword8

-- label=20
::label::

-- identifier substyles.11.1=128
moon
